### PR TITLE
Fix markup alignment inconsistency

### PR DESCRIPTION
**Pull request by danx91**
## tl;dr
Changed text position calculation, so it is aligned as expected while using center or right text alignment.

## Description
Currently, markup object is aligned according to it's size (`self.totalWidth`) but text is aligned according to maximum size (`self.maxWidth`). This results in misaligned text while using center or right alignment. 
offcentered
_Image of current markup behavior with `halign` and `textAlign` both set to `TEXT_ALIGN_CENTER` (white line - center of frame, green box - size of markup object, red box - max size of markup object)._

As visible on the image, the markup object itself is aligned correctly, but the text is off-centered. The same problem occurs while using right alignment, making the text aligned to the right side of the red box.

correct
_Image of changed markup behavior with `halign` and `textAlign` both set to `TEXT_ALIGN_CENTER` (white line - center of frame, green box - size of markup object, red box - max size of markup object)._

Now the text is also aligned according to actual size, resulting in text aligned perfectly in the center of the frame, and it also works correctly with `TEXT_ALIGN_RIGHT`.

## Possible backward compatibility issues
Some developers may have corrected misaligned markup on their own, so changing this will cause the markup text to be 'corrected twice' and making it off-centered for them. However, in my opinion, current behavior is wrong and should be changed to work as expected - centering both markup object and text should produce a perfectly centered result.

### CODE REVIEW

**Review by robotboy655 — APPROVED** [review]
Seems the issue was introduced by https://github.com/Facepunch/garrysmod/pull/1732 when the text align argument was added.
